### PR TITLE
cama-1403: referral features

### PR DESCRIPTION
**Pull request by julianbachiller**
Add support for the new Integration API Referral features.

The client methods for referral codes creation and search profiles by attributes are added.

### CODE REVIEW

File: lib/integration/client.rb [hunk 1]
```diff
@@ -51,6 +51,22 @@ def close_customer_session(session_id)
         update_customer_session session_id, { state: "closed" }
       end
 
+      def create_referral_code(campaign_id, advocate_profile_id, friend_profile_id, start, expire)
```
Comment (grncdr) [inline lib/integration/client.rb]: Please use an options object for the params that are actually optional.

File: lib/integration/referral_code.rb [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+      def code
+        @raw["code"]
+      end
+
+      def friend_profile_integration_id?
+        @raw["friendProfileIntegrationId"]
+      end
+
+      def start_date?
```
Comment (grncdr) [inline lib/integration/referral_code.rb]: this and `expiry_date?` should not have the question mark at the end. Only use a question mark for methods that return a boolean.

File: lib/integration/referral_code_options.rb [hunk 3]
```diff
@@ -0,0 +1,21 @@
+module TalonOne
+  module Integration
+    class ReferralCodeOptions
+      def initialize(raw_data)
+        @raw = raw_data
+      end
+
+      def friend_id?
```
Comment (grncdr) [inline lib/integration/referral_code_options.rb]: again, no question mark on these methods.

File: lib/integration/referral_code.rb [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+
+      def campaign_id
+        @raw["campaignId"]
+      end
+
+      def created
+        @raw["created"]
+      end
+
+      def advocate_profile_integration_id
```
Comment (grncdr) [inline lib/integration/referral_code.rb]: can we shorten this method name to `advocate_id`?

File: lib/integration/referral_code.rb [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+        @raw["advocateProfileIntegrationId"]
+      end
+
+      def code
+        @raw["code"]
+      end
+
+      def friend_profile_integration_id?
```
Comment (grncdr) [inline lib/integration/referral_code.rb]: and `friend_id` here.

File: lib/integration/search_profiles_result.rb [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+
+module TalonOne
+  module Integration
+    class SearchProfilesResult
+      def initialize(raw_data)
+        @raw = raw_data
+      end
+
+      def profiles
+        @profiles ||= @raw.map do |raw_array|
```
Comment (grncdr) [inline lib/integration/search_profiles_result.rb]: This should be `@raw["data"].map ...`

File: lib/integration/referral_code.rb (hunk 2 above)
Comment (julianbachiller) [inline lib/integration/referral_code.rb]: Thx, I followed the wrong convention, thought it was for _nil_ -able fields :)

**Pull request merged**